### PR TITLE
Ollama server test

### PR DESCRIPTION
**Pull request by santhiya-mj**
Added new test case for OllamaServer functionality.

### CODE REVIEW

File: .github/workflows/integrationtests.yml
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:1]: This code is already commited? 

Why do we need this file in pull request?

File: test/IntegrationTests/test_ollamaserver.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            "model": self.valid_model,
+            "prompt": self.valid_prompt,
+            "stream": False
+        }
+        response = self.send_post_request(self.generate_endpoint, payload)
+
+        self.assertEqual(response.status_code, 200, "Expected status code 200.")
+        response_json = response.json()
+        self.assertIn("response", response_json, "Response JSON should contain 'response' key.")
+        self.assertIn("Paris", response_json["response"], "Expected response to contain 'Paris'.")
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_ollamaserver.py:61]: Just check whether we are getting output. Particularly 'Paris' word might fail

You can change it to checking if it contains 'Paris', it succeeds

If it does not contain Paris check whether the model has outputted some text and if yes, that should be ok

File: test/IntegrationTests/test_ollamaserver.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        logger.info("Test Case Passed 1: Valid prompt generated a correct response.\n")
+
+    # Negative Test Cases
+    def test_generate_missing_prompt(self):
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_ollamaserver.py:65]: We can delete this testcase

File: .github/workflows/integrationtests.yml
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml:1]: I accidentally added a file to this branch. As a result, when I created the pull request, the integration.yml file was also committed to the PR.

File: test/IntegrationTests/test_ollamaserver.py (hunk 1 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_ollamaserver.py]: I've done this part, sir. I modified the code.

File: test/IntegrationTests/test_ollamaserver.py (hunk 2 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_ollamaserver.py]: I deleted the test case from this file

File: .github/workflows/integrationtests.yml [hunk 3]
```diff
@@ -28,7 +28,23 @@ jobs:
           PROJECT_ROOT: ${{ github.workspace }}
         run: docker-compose up -d
 
-      # Step 4: Test if the server is up
+      # Step 4: Check if Ollama server is up
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:31]: Lets remove this file from the pull request
.github/workflows/integrationtests.yml

**Pull request closed without merging**